From 32d563517e9571b980421419fbebb342c3b4efdc Mon Sep 17 00:00:00 2001
From: Lu Jialin <lujialin4@huawei.com>
Date: Mon, 14 Nov 2022 10:09:43 +0800
Subject: [PATCH 0039/1398] [Huawei] cgroup: add config isolation for
 cgroup_kill in cgroupv1

Offering: HULK
hulk inclusion
category: feature
bugzilla: 187970

-------------------------------

Introduce CONFIG_CGROUP_V1_KILL to isolate cgroup_kill feature in
cgroupv1.

Signed-off-by: Lu Jialin <lujialin4@huawei.com>
---
 arch/arm64/configs/openeuler_defconfig | 1 +
 arch/x86/configs/openeuler_defconfig   | 1 +
 init/Kconfig                           | 5 +++++
 kernel/cgroup/cgroup-v1.c              | 2 ++
 4 files changed, 9 insertions(+)

diff --git a/arch/arm64/configs/openeuler_defconfig b/arch/arm64/configs/openeuler_defconfig
index be3c51633c4b..0a9bff266c43 100644
--- a/arch/arm64/configs/openeuler_defconfig
+++ b/arch/arm64/configs/openeuler_defconfig
@@ -129,6 +129,7 @@ CONFIG_ARCH_SUPPORTS_INT128=y
 CONFIG_NUMA_BALANCING=y
 CONFIG_NUMA_BALANCING_DEFAULT_ENABLED=y
 CONFIG_CGROUPS=y
+CONFIG_CGROUP_V1_KILL=y
 CONFIG_PAGE_COUNTER=y
 CONFIG_MEMCG=y
 CONFIG_MEMCG_SWAP=y
diff --git a/arch/x86/configs/openeuler_defconfig b/arch/x86/configs/openeuler_defconfig
index 812b061ff1da..13c3938b42d0 100644
--- a/arch/x86/configs/openeuler_defconfig
+++ b/arch/x86/configs/openeuler_defconfig
@@ -148,6 +148,7 @@ CONFIG_ARCH_SUPPORTS_INT128=y
 CONFIG_NUMA_BALANCING=y
 CONFIG_NUMA_BALANCING_DEFAULT_ENABLED=y
 CONFIG_CGROUPS=y
+CONFIG_CGROUP_V1_KILL=y
 CONFIG_PAGE_COUNTER=y
 CONFIG_MEMCG=y
 CONFIG_MEMCG_SWAP=y
diff --git a/init/Kconfig b/init/Kconfig
index d8fe58b346c4..73334071fa32 100644
--- a/init/Kconfig
+++ b/init/Kconfig
@@ -1215,6 +1215,11 @@ config CGROUP_FILES
 	  This supports catching misbehaving processes and
 	  return EMFILE instead of ENOMEM for kernel memory limits.
 
+config CGROUP_V1_KILL
+	bool "Kill All Tasks In Cgroup"
+	default n
+	depends on CGROUPS
+
 endif # CGROUPS
 
 menuconfig NAMESPACES
diff --git a/kernel/cgroup/cgroup-v1.c b/kernel/cgroup/cgroup-v1.c
index be884bc2ae61..36bc1c6a977b 100644
--- a/kernel/cgroup/cgroup-v1.c
+++ b/kernel/cgroup/cgroup-v1.c
@@ -660,11 +660,13 @@ struct cftype cgroup1_base_files[] = {
 		.write = cgroup_release_agent_write,
 		.max_write_len = PATH_MAX - 1,
 	},
+#ifdef CONFIG_CGROUP_V1_KILL
 	{
 		.name = "cgroup.kill",
 		.flags = CFTYPE_NOT_ON_ROOT,
 		.write = cgroup_kill_write,
 	},
+#endif
 	{ }	/* terminate */
 };
 
-- 
2.17.1

